torch/distributed/rpc/api.py

```diff
@@ -498,29 +499,34 @@ def method(self, *args, **kwargs):
     return method
 
 
-for method_name, method in inspect.getmembers(PyRRef):
-    # Ignore magic methods, except "__str__".
-    if method_name.startswith("_") and method_name != "__str__":
-        continue
+def _update_PyRRef_docstrings():
+    for method_name, method in inspect.getmembers(PyRRef):
+        # Ignore magic methods, except "__str__".
+        if method_name.startswith("_") and method_name != "__str__":
+            continue
 
-    # Get pybind11 generated docstring.
-    # It's like,
-    """
-    to_here(self: torch.distributed.rpc.PyRRef, timeout: float=-1.0) -> object
+        # Get pybind11 generated docstring.
+        # It's like,
+        """
+        to_here(self: torch.distributed.rpc.PyRRef, timeout: float=-1.0) -> object
 
-        Blocking call that copies the value of the RRef from the owner
-        to the local node and returns it. If the current node is the
-        owner, returns a reference to the local value.
-    """
-    docstring = getattr(method, "__doc__", None)
-    assert docstring is not None, "RRef user-facing methods should all have docstrings."
+            Blocking call that copies the value of the RRef from the owner
+            to the local node and returns it. If the current node is the
+            owner, returns a reference to the local value.
+        """
+        docstring = getattr(method, "__doc__", None)
+        assert docstring is not None, "RRef user-facing methods should all have docstrings."
+
+        # Do surgery on pybind11 generated docstrings.
+        docstring = docstring.replace("torch.distributed.rpc.PyRRef", "torch.distributed.rpc.RRef")
+
+        # Attach user-facing RRef method with modified docstring.
+        new_method = _method_factory(method_name, docstring)
+        setattr(RRef, method_name, new_method)
 
-    # Do surgery on pybind11 generated docstrings.
-    docstring = docstring.replace("torch.distributed.rpc.PyRRef", "torch.distributed.rpc.RRef")
 
-    # Attach user-facing RRef method with modified docstring.
-    new_method = method_factory(method_name, docstring)
-    setattr(RRef, method_name, new_method)
+if sysconfig.get_config_var("WITH_DOC_STRINGS"):
```

Comment:
Is this setting also properly set to True when running with `python -OO` ?
If not, I think it is fine to remove the assert above and just skip updating the docstring when none is present. We have CI checks that ensure documentations are properly done now.

Reply:
Removing the assert and skipping the docstring update if it is not there sounds like the most reliable way to fix this. It would be great, thanks!